include/Core/Posts/Metabox/DeletePostsByPostTypeMetabox.php

```diff
@@ -0,0 +1,151 @@
+		$this->messages      = array(
+			'box_label' => __( 'By Custom Post Type', 'bulk-delete' ),
+			'scheduled' => __( 'The selected posts are scheduled for deletion', 'bulk-delete' ),
+		);
+	}
+
+	public function render() {
+		$types_array = array();
+
+		$types = get_post_types( array(
+				'_builtin' => false,
+			), 'names'
+		);
+
+		if ( count( $types ) > 0 ) {
+			foreach ( $types as $type ) {
+				$post_count = wp_count_posts( $type );
+				if ( $post_count->publish > 0 ) {
+					$types_array["$type-publish"] = $post_count->publish;
+				}
+				if ( $post_count->future > 0 ) {
+					$types_array["$type-future"] = $post_count->future;
+				}
+				if ( $post_count->pending > 0 ) {
+					$types_array["$type-pending"] = $post_count->pending;
+				}
+				if ( $post_count->draft > 0 ) {
+					$types_array["$type-draft"] = $post_count->draft;
+				}
+				if ( $post_count->private > 0 ) {
+					$types_array["$type-private"] = $post_count->private;
+				}
+			}
+		}
+
+		if ( count( $types_array ) > 0 ) {
+?>
+            <!-- Custom post type Start-->
+            <h4><?php _e( 'Select the custom post types from which you want to delete posts', 'bulk-delete' ) ?></h4>
+
+            <fieldset class="options">
+            <table class="optiontable">
+<?php
+			foreach ( $types_array as $type => $count ) {
+?>
+                <tr>
+                    <td scope="row" >
+                        <input name="smbd_types[]" value="<?php echo $type; ?>" type="checkbox">
+                    </td>
+                    <td>
+						<label for="smbd_types"><?php echo $this->display_post_type_status( $type ), ' (', $count, ')'; ?></label>
+                    </td>
+                </tr>
+<?php
+			}
+
+			$this->render_filtering_table_header();
+			$this->render_restrict_settings();
+			$this->render_delete_settings();
+			$this->render_limit_settings();
+			$this->render_cron_settings();
+?>
+            </table>
+            </fieldset>
+<?php
+			$this->render_submit_button();
+		} else {
+            printf( '<h4>%s</h4>', __( "This WordPress installation doesn't have any non-empty custom post types", 'bulk-delete' ) );
+		}
+	}
+
+	protected function convert_user_input_to_options( $request, $options ) {
+		$options                   = array();
+
+		$options['selected_types'] = bd_array_get( $request, 'smbd_types' );
+		$options['restrict']       = bd_array_get_bool( $request, 'smbd_types_restrict', false );
+		$options['limit_to']       = absint( bd_array_get( $request, 'smbd_types_limit_to', 0 ) );
+		$options['force_delete']   = bd_array_get_bool( $request, 'smbd_types_force_delete', false );
+
+		$options['date_op']        = bd_array_get( $request, 'smbd_types_op' );
+		$options['days']           = absint( bd_array_get( $request, 'smbd_types_days' ) );
+
+		return $options;
+	}
+
+	public function delete( $delete_options ) {
+		// Backward compatibility code. Will be removed in Bulk Delete v6.0
```

Comment:
We can replace the `delete` method with just the `build_query` method.

Look into either `DeletePostsByCategoryMetabox` or  `DeletePostsByTagMetabox` to see how to do it. This is something that I recently added.